### DISCUSSION TITLE
Relation Query

### DISCUSSION THREAD

**Discussion by Eggtron** [pocketbase/pocketbase#1641] [Q&A] [answered]
Hello all,

Sorry if my question is stupid. 

Table for "Kundenobjekte"
```json
{
      "id": "RECORD_ID",
      "collectionId": "t6sib33ajk21ppe",
      "collectionName": "kundenobjekt",
      "created": "2022-01-01 01:00:00.123Z",
      "updated": "2022-01-01 23:59:59.456Z",
      "kundenobjekttyp": "RELATION_RECORD_ID",
      "kunden_id": "RELATION_RECORD_ID",
      "strasse": "test",
      "postleitzahl": "test",
      "stadt": "test"
    },
```
Table for "global_kundenobjekte"
```json
{
      "id": "RECORD_ID",
      "collectionId": "dmrp06iubk4mgzc",
      "collectionName": "global_kundenobjekttyp",
      "created": "2022-01-01 01:00:00.123Z",
      "updated": "2022-01-01 23:59:59.456Z",
      "text": "test"
    },
```

How could i fetch a list of "Kundenobjekt", filtered by "kunden_id" and expanded to "global_kundenobjekt"? :)

```js
  async function getkundenobjektliste() {
    try {
      
      const records = await pb.collection("kundenobjekt").getFullList(
        200,
        50,
        {
          filter: 'kunden_id ~ "${kundenid}"',
        },
        { expand: "global_kundenobjekttyp" }
      );
      rows = records;
      console.log(records);
    } catch (e) {
      console.log(e);
    }
  }
```

I tried a ton of different ways to write it, but as a beginner i am now just at the limit of my knowledge. 
API Rules are fine. 

By the way, PB Backend v0.10.4 and js sdk 0.8.4

**Comment by ganigeorgiev** [comment, marked as answer]
You can check https://pocketbase.io/docs/expanding-relations/.

I'm not sure that I understand the above example. For direct relation expansion you have to use the relation field name. `global_kundenobjekttyp` in the above expand doesn't seem to exist as a relation field in the `kundenobjekt` collection (maybe that is the field `kundenobjekttyp`?).

Also note that the request client must satisfy the relation's View API rule in order to be able to expand the related record(s).

**Reply by Eggtron** [reply]
Good morning again, 

Thank you for your effort, PB is awesome. Finally you are right and my code was wrong too...
